policy/policy_handler/subscription.go

```diff
@@ -0,0 +1,36 @@
+package policy_handler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/atomist-skills/go-skill"
+	"github.com/atomist-skills/go-skill/policy/data"
+	"olympos.io/encoding/edn"
+)
+
+func WithSubscription() Opt {
+	return func(h *EventHandler) {
+		h.subscriptionDataProviders = append(h.subscriptionDataProviders, getSubscriptionData)
+		h.dataSourceProviders = append(h.dataSourceProviders, getSyncDataSources)
+	}
+}
+
+func getSubscriptionData(ctx context.Context, req skill.RequestContext) ([][]edn.RawMessage, skill.Configuration, error) {
+	if req.Event.Context.Subscription.Name == "" {
+		return nil, skill.Configuration{}, nil
+	}
+
+	return req.Event.Context.Subscription.Result, req.Event.Context.Subscription.Configuration, nil
+}
+
+func getSyncDataSources(ctx context.Context, req skill.RequestContext) ([]data.DataSource, error) {
+	gqlDs, err := data.NewSyncGraphqlDataSource(ctx, req)
```

Comment:
I think all handlers should support sync queries, not just subscriptions.

Reply:
I haven't gated this behind any event name checks so it should be available any time that it is selected in the ChainDataSource, although there's not currently a way to specify which data source you want a query to use. As-is, the sync one is more of a debug/fallback option. Can you think of a use case where we would _prefer_ to use a sync query instead of an async one?

Reply:
The main use cases for sync are probably:

- Local eval
- When multiple queries are required
- Fallback

Now that I think more about it, I wonder if we want to separate sync out into it's own thing and make it an explicit opt-in, rather than adding it as part of `subscription`...

Reply:
see below conversation, done in b330d7f437a91082b7285433f235e82c313cbb25